test/test.go

```diff
@@ -122,19 +122,23 @@ func (req *Request) Validate() error {
 		}
 		return nil
 	}
-	return fmt.Errorf("Unsupported method \"%v\"", req.Method)
+	return fmt.Errorf("unsupported method \"%v\"", req.Method)
 }
 
-func getPort(u *url.URL) int32 {
+func getPort(u *url.URL) (int32, error) {
 	p := u.Port()
 	if p == "" {
 		if u.Scheme == "https" {
-			return 443
+			return 443, nil
 		}
-		return 80
+		return 80, nil
 	}
-	portnum, _ := strconv.Atoi(p)
-	return int32(portnum)
+	portnum, err := strconv.ParseInt(p, 10, 32)
```

Comment:
The port is already validated in the `URL` object creation, so this should be unreachable.